src/analysis/ConfigurationExt.cpp

```diff
@@ -91,18 +91,51 @@ namespace espressopp {
       //end = coordinates.end();
     }
 
-    int ConfigurationExtIterator::nextId()
+    size_t ConfigurationExtIterator::currentId()
+    {
+      if (it == end) {
+        PyErr_SetString(PyExc_StopIteration, "No more data.");
+        boost::python::throw_error_already_set();
+      }
+
+      size_t id = (*it).first;
+      return id;
+    }
+
+    RealND ConfigurationExtIterator::currentProperties()
     {
       if (it == end) {
         PyErr_SetString(PyExc_StopIteration, "No more data.");
         boost::python::throw_error_already_set();
       }
 
-      int id = (*it).first;
+      RealND props = (*it).second;
+      return props;
+    }
+
+    void ConfigurationExtIterator::incrementIterator()
+    {
+      /*
+      if (it == end) {
```

Comment:
This check is done also when the particle loop in DumpXYZ.cpp is entered, so i commented here, but obviously makes more sense to check at this point before the checks inside the loop for the next particle. My suggestion is to return a bool from ConfigurationExtIterator::incrementIterator() and have a check in DumpXYZ.cpp for that and exit the particle loop so there's no need for an extra comparison later. Thoughts?